### PR TITLE
Simulated annealing and metropolis algorithms

### PR DESCRIPTION
**Pull request by ahrajabi**
This branch is going to add the simulated annealing algorithm (or Metropolis algorithm for the case of fixed temperature) to the project.

### CODE REVIEW

File: src/experiments/experiments.jl [hunk 1]
```diff
@@ -175,6 +184,9 @@ end
 
 
 function mkworkspace(exp::Experiment)
+    if !exp.save
+        return
+    end
```
Comment (journeyer) [inline src/experiments/experiments.jl]: I believe it is unnecessary due to [this if statement](https://github.com/Pidoh/Pidoh.jl/pull/16/files#diff-c604085a8194a47bc3c034dbdedac758c2bde3894a08454bf5e2fd49303e843eR74-R76).

File: src/experiments/experiments.jl [hunk 2]
```diff
@@ -202,6 +214,9 @@ function mkworkspace(exp::Experiment)
 end
 
 function updateworkspace(exp::Experiment)
+    if !exp.save
+        return
+    end
```
Comment (journeyer) [inline src/experiments/experiments.jl]: IMHO, `updateworkspace` and `mkworkspace` should have the same structure so, remove this if statement if you did it for `mkworkspace` too.

File: src/experiments/experiments.jl (hunk 1 above)
Comment (ahrajabi) [reply src/experiments/experiments.jl]: Thanks! removed.

File: src/experiments/experiments.jl (hunk 2 above)
Comment (ahrajabi) [reply src/experiments/experiments.jl]: Done!

**Pull request merged**